src/MasterXaction.h

```diff
@@ -41,7 +41,24 @@ class MasterXaction : public RefCountable
 public:
     typedef RefCount<MasterXaction> Pointer;
 
-    explicit MasterXaction(const XactionInitiator anInitiator) : initiator(anInitiator) {};
+    /// Does not have association with a listening port.
+    /// \param anInitiator any value other from XactionInitiator::initClient.
+    explicit MasterXaction(const XactionInitiator anInitiator) :
+        initiator(anInitiator)
+    {
+        assert(!initiator.in(XactionInitiator::initClient));
```

Comment:
PR commit a7e9f64 replaces this dangerous runtime assertion with a compile-time one. I hope that the replacement and the correction below will address this change request.

> Amos: This will block using MasterXaction for transactions like HTCP which are initiated by a remote client, but where no SquidPort object is available yet

The assertion does not "block" portless HTCP transactions and alike (e.g., src/htcp.cc compiles fine after converting this runtime assert() to a static_assert in a7e9f64). I suspect that you were thinking that initClient includes HTCP transactions. It does not. We have initHtcp for them. Our initClient is limited to FTP and HTTP transactions that do have port information. Here is a quote from the official documentation, mentioning both "htcp" (i.e. initHtcp) and "client" (i.e. initClient):

```
	  # Supported initiators are:
	  #  htcp: matches HTCP requests from peers
	  #  icp: matches ICP requests to peers
      ...
	  #  client: matches transactions containing an HTTP or FTP
	  #     client request received at a Squid *_port
```